File: Temas/Ordenamiento/insercion.js
const Arreglo = [
  { contenedor: { dato: 5 } },
  { contenedor: { dato: 7 } },
  { contenedor: { dato: 10 } },
  { contenedor: { dato: 2 } },
  { contenedor: { dato: 3 } },
  { contenedor: { dato: 1 } },
];

function insercion(arreglo) {
  let pivote;
  let j;
  for (let i = 0; i < arreglo.length; i++) {
    pivote = arreglo[i];
    j = i - 1;
    while (j >= 0 && arreglo[j].contenedor.dato > pivote.contenedor.dato) {
      arreglo[j + 1] = arreglo[j];
      j--;
    }
    arreglo[j + 1] = pivote;
  }
  return arreglo;
}

console.log(insercion(Arreglo));

File: Temas/Ordenamiento/insercion.c
#include <stdio.h>

typedef struct {
  int dato;
}Dato;

typedef struct {
  Dato generico;    
}TDA;

void insercion(TDA arreglo[], int tam){
  TDA pivote;
  int j;
  for(int i = 0; i < tam; i++){
    pivote = arreglo[i];
    j = i-1;
    while(j >=0 && arreglo[j].generico.dato > pivote.generico.dato){
      arreglo[j+1] = arreglo[j];
      j--;
    }
    arreglo[j+1] = pivote;
  }
}

int main() {
  TDA arreglo[5] = {
    {5},
    {8},
    {1},
    {10},
    {3}
  };

  insercion(arreglo, 5);

  for(int i = 0; i<5; i++){
    printf("%d, ", arreglo[i].generico.dato);
  }
}